packages/dotlottie-js/src/common/dotlottie-common.ts

```diff
@@ -245,18 +246,33 @@ export class DotLottieCommon {
     });
   }
 
-  private _renameImageAssets(): void {
-    const images: Map<string, LottieImageCommon[]> = new Map();
+  /**
+   * Generates a map of duplicate image ids and their count.
+   * @returns Map of duplicate image ids and their count.
+   */
+  private _getMapOfDuplicateImageIds(): Map<string, number> {
+    let count = 0;
+    const dupeMap = new Map<string, number>();
 
     this.animations.forEach((animation) => {
-      images.set(animation.id, animation.imageAssets);
+      animation.imageAssets.forEach((imageAsset) => {
+        if (dupeMap.has(imageAsset.id)) {
+          count = dupeMap.get(imageAsset.id) ?? 0;
+          dupeMap.set(imageAsset.id, count + 1);
+        } else {
+          dupeMap.set(imageAsset.id, 1);
```

Comment:
This seems to be an `occuranceMap` not `duplicateMap`, right ? as based on this line, even if there is a unique image id, still would be pushed to the map with an occurance of `1`

Reply:
It was duplicateMap as duplicate filename not the content of the image, will change to filename occurence map